File: files/200-299/235/CH-235.py
import pandas as pd

path = "200-299/235/CH-235 Filtering & Removing.xlsx"
input = pd.read_excel(path, usecols="B:F", skiprows=1, nrows=5)
test = pd.read_excel(path, usecols="H", skiprows=1, nrows=4).rename(columns=lambda col: col.replace('.1', ''))

input_long = input.melt(id_vars=input.columns[0], value_name="Value")
grouped = input_long.groupby(input.columns[0])["Value"].apply(lambda x: ",".join(map(str, sorted(x, reverse=True))))
result = grouped.drop_duplicates().index.to_frame(index=False)


print(result.equals(test)) # True

File: files/200-299/235/CH-235.R
library(tidyverse)
library(readxl)

path = "files/200-299/235/CH-235 Filtering & Removing.xlsx"
input = read_excel(path, range = "B2:F7")
test = read_excel(path, range = "H2:H6")

result = input %>%
  pivot_longer(-1, values_to = "Value") %>%
  summarise(nums = toString(sort(Value, decreasing = TRUE)), .by = ID) %>%
  distinct(nums, .keep_all = TRUE) %>%
  select(ID)

all.equal(result, test)
#> [1] TRUE
